soc/xtensa/intel_adsp/tools/cavstwist.sh

```diff
@@ -4,122 +4,93 @@
-# can't find the test directory!  And worse, the scripts are
-# asynchronous and may start in either order, but our console script
-# can't be allowed to run until after the flash.  If it does, it will
-# pull old data (from the last test run!)  out of the buffer and
-# confuse twister.
 #
-# So the solution here is to have the console script do the trace
-# reading AND the flashing.  The flash script merely signs the binary
-# and places it into ZEPHYR_BASE for the console script to find.  The
-# console script then just has to wait for the binary to appear (which
-# solves the ordering problem), flash it, delete it (so as not to
-# confuse the next test run), and emit the adsplog output as expected.
+# Other notes:
+# The signing key must be correct for your device.  In general we want
+# to be using the SOF-curated "community" keys.  Note that cAVS 2.5
+# uses a different (longer) key than the earlier platforms. You can
+# change the key using in the tree:
+#  $ZEPHYR_BASE/board/xtensa/{cavs board}/board.cmake
+#
+#  set(RIMAGE_SIGN_KEY otc_private_key_3k.pem)
 #
-# And notice the race protection: it's possible twister will run a
-# signing script IMMEDIATELY, simultaneous with the last flash script.
-# This may thus clobber the version that the other script hasn't seen
-# yet.  So we add an extra wait step in the signing path for the
-# loader to "accept" the file.
+# And put the key you wanna use in:
+#  moudles/audio/sof/rimage/config
+#
+# The rimage signed tool must be in place, checked out and built (!)
+# rimage tool from SOF: https://github.com/thesofproject/rimage
 #
 ########################################################################
 
-# Three configuration variables to set.  Have to be passed via the
-# environment because this script is launched from twister.
+# The first passed argument for this script from west flash is the
+# build directory. The second argument is the remote CAVS host.
+CAVS_RIMAGE=$1/zephyr/zephyr.ri
+CAVS_HOST=$2
 
-if [ -z "$CAVS_HOST" -o -z "$CAVS_KEY" -o -z "$CAVS_RIMAGE" ]; then
+if [ ! -f "$CAVS_RIMAGE" ]; then
+    echo "  NO rimage found. Abort running." 1>&2
+fi
+
+# If the CAVS_HOST was not passing by west flash command, it will try
+# again to serach the environment variables ZEPHYR_CAVS_HOST.
+if [ -z "$CAVS_HOST"]; then
+    CAVS_HOST=$ZEPHYR_CAVS_HOST
+fi
+
+if [ -z "$CAVS_HOST"]; then
     echo "Missing cavstwist.sh configuration, must have:" 1>&2
+    echo "  CAVS_HOST as the first argument." 1>&2
+    echo "  Usage: west flash [CAVS_HOST]" 1>&2
+    echo "  OR you can also do:" 1>&2
     echo "  export CAVS_HOST=ssh_host_name" 1>&2
-    echo "  export CAVS_KEY=/path/to/signing_key.pem" 1>&2
-    echo "  export CAVS_RIMAGE=/path/to/built/rimage/dir" 1>&2
+    echo "  before you run west flash." 1>&2
+    echo "  Abort running." 1>&2
     exit 1
 fi
 
 ########################################################################
 
 CAVSTOOL=$ZEPHYR_BASE/soc/xtensa/intel_adsp/tools/cavstool.py
```

Comment:
So this is a west runner, that calls a shell script, that calls a python file?

This is kind of a mess in my opinion.

Reply:
Thanks for your review, @mbolivar-nordic . Would you mean it's better to use python script instead of shell script here?

Reply:
What I mean is that the extra indirection looks like suspect design to me. It looks like cavstool.py was added by @andyross in f581d59821f34ad2c1f320dc3597a0883e7c6792, and then cavstwist.sh was added by the same person in e71b1a1148c7ac176a56a1800162e954cb9d8eae specifically as a twister integration script, and now there is a west runner (python) that's calling  the twister integration script (shell) that's calling cavstool.py (python). That doesn't make much sense.

Reply:
Hi @mbolivar-nordic , I tried to make a dedicate west runner for it. It will call the cavstool.py (firmware loader and runner in the DUT host side). Could you please help to see if this implementation make sense or not, when you got a change, thank you!